### PR TITLE
CONCF-1058 | add modal component to curated content editor and style it

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: front/styles/main/component/_modal-dialog.scss [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
+	color: $semi-dark;
+	display: flex;
+	font-size: 16px;
+	padding: 20px;
 	position: fixed;
 	top: 0;
 	width: 100%;
-	z-index: $z-modal-dialog;
-}
+	z-index: $z-1;
```
Comment (rogatty) [inline front/styles/main/component/_modal-dialog.scss]: Use https://github.com/Wikia/mercury/blob/dev/front/styles/main/_wikia-variables.scss

File: front/scripts/main/components/CuratedContentEditorItemFormComponent.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ App.CuratedContentEditorItemFormComponent = Em.Component.extend(
 		imageWidth: 300,
 		maxLabelLength: 48,
 		debounceDuration: 250,
+		imageModalVisible: false,
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.ts]: I'd call it `imageMenuVisible`

File: front/templates/main/components/curated-content-editor-item-form.hbs [hunk 3]
```diff
@@ -11,13 +11,12 @@
 		{{/if}}
 		{{loading-spinner active=isLoading overlay=false}}
 	</div>
-	<div class="image-modal-wrapper {{if imageModalVisible 'visible'}}" >
-		<div class="image-modal-layover" {{action 'hideImageModal'}}></div>
-		<ul class="image-modal">
+	{{#modal-dialog isVisible=imageModalVisible type="menu"}}
```
Comment (rogatty) [inline front/templates/main/components/curated-content-editor-item-form.hbs]: Single quotes

File: front/styles/main/component/_modal-dialog.scss [hunk 4]
```diff
@@ -1,27 +1,52 @@
-$modal-dialog-margin: 1em;
+$light-blue: #1a5eb8;
```
Comment (Warkot) [inline front/styles/main/component/_modal-dialog.scss]: `$color-blue-light` defined in `front/vendor/wikia-style-guide/gh-pages/css/main.scss`

File: front/styles/main/component/_modal-dialog.scss [hunk 5]
```diff
@@ -1,27 +1,52 @@
...
+$semi-dark: #39424d;
```
Comment (Warkot) [inline front/styles/main/component/_modal-dialog.scss]: `$color-gray-dark`

**Comment by Warkot** [comment]
Some scss-lint warnings:
* [front/styles/main/component/_curated-content-editor.scss:204](https://github.com/Wikia/mercury/blob/CONCF-1058/front/styles/main/component/_curated-content-editor.scss#L204) [W] ColorVariable: Color literals like `#000` should only be used in variable declarations; they should be referred to via variable everywhere else.
* [front/styles/main/component/_curated-content-editor.scss:259](https://github.com/Wikia/mercury/blob/CONCF-1058/front/styles/main/component/_curated-content-editor.scss#L259) [W] ImportantRule: !important should not be used
* [front/styles/main/component/_curated-content-editor.scss:310](https://github.com/Wikia/mercury/blob/CONCF-1058/front/styles/main/component/_curated-content-editor.scss#L310) [W] BorderZero: `border: 0 is preferred over `border: none`

**Comment by bognix** [comment]
Important is there for a reason. I can fix remaining two

**Comment by Warkot** [comment]
LGTM :+1:

**Comment by tomasznapieralski** [comment]
Big LGTM

**Pull request merged**